src/common/src/com/microsoftopentechnologies/tooling/msservices/helpers/azure/rest/RestServiceManagerBaseImpl.java

```diff
@@ -0,0 +1,207 @@
+        }
+    }
+
+    @NotNull
+    public HttpsURLConnection getSSLConnection(@NotNull String managementUrl,
+                                               @NotNull String path,
+                                               @NotNull ContentType contentType)
+            throws AzureCmdException {
+        try {
+            URL myUrl = new URL(managementUrl + path);
+            HttpsURLConnection conn = (HttpsURLConnection) myUrl.openConnection();
+            conn.addRequestProperty(USER_AGENT_HEADER, getPlatformUserAgent());
+            conn.addRequestProperty(TELEMETRY_HEADER, getPlatformUserAgent());
+            conn.addRequestProperty(X_MS_VERSION_HEADER, AZURE_API_VERSION);
+            conn.addRequestProperty(ACCEPT_HEADER, "");
+            conn.addRequestProperty(CONTENT_TYPE_HEADER, contentType.toString());
+
+            return conn;
+        } catch (IOException e) {
+            throw new AzureCmdException(e.getMessage(), e);
+        }
+    }
+
+    @NotNull
+    protected static HttpResponse getResponse(@NotNull final String method,
+                                              @Nullable final String postData,
+                                              @NotNull HttpsURLConnection sslConnection)
+            throws IOException, AzureCmdException {
+        try {
+            sslConnection.setRequestMethod(method);
+            sslConnection.setDoOutput(postData != null);
+
+            if (postData != null) {
+                DataOutputStream wr = new DataOutputStream(sslConnection.getOutputStream());
+                wr.writeBytes(postData);
+                wr.flush();
+                wr.close();
+            }
+
+            return getResponse(sslConnection);
+        } finally {
+            sslConnection.disconnect();
+        }
+    }
+
+    @NotNull
+    private static String getPlatformUserAgent() {
+        String version = DefaultLoader.getPluginComponent().getSettings().getPluginVersion();
+        return String.format(
+                "%s/%s (lang=%s; os=%s; version=%s)",
+                DefaultLoader.PLUGIN_ID,
+                version,
+                "Java",
+                System.getProperty("os.name"),
+                version);
+    }
+
+    @NotNull
+    private static HttpResponse getResponse(@NotNull HttpsURLConnection sslConnection)
+            throws IOException {
+        int code = sslConnection.getResponseCode();
+        String message = Strings.nullToEmpty(sslConnection.getResponseMessage());
+        Map<String, List<String>> headers = sslConnection.getHeaderFields();
+
+        if (headers == null) {
+            headers = new HashMap<String, List<String>>();
+        }
+
+        InputStream is;
+        String content;
+
+        try {
+            is = sslConnection.getInputStream();
+        } catch (IOException e) {
+            is = sslConnection.getErrorStream();
+        }
+
+        if (is != null) {
+            content = readStream(is);
```

Comment:
I'd recommend replacing `readStream` with [`CharStreams.toString`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/io/CharStreams.html#toString(java.lang.Readable)) from Google Guava. Also, we should probably close `is` from a `finally` block.